msbuild/Xamarin.MacDev.Tasks/Tasks/FindAotCompilerTaskBase.cs

```diff
@@ -1,160 +1,47 @@
+</Target>";
+
 			if (MonoAotCrossCompiler?.Length > 0 && string.IsNullOrEmpty (Environment.GetEnvironmentVariable ("XAMARIN_FORCE_AOT_COMPILER_PATH_COMPUTATION"))) {
 				var aotCompilerItem = MonoAotCrossCompiler.SingleOrDefault (v => v.GetMetadata ("RuntimeIdentifier") == RuntimeIdentifier);
 
 				if (aotCompilerItem is null) {
 					Log.LogMessage (MessageImportance.Low, "Unable to find the AOT compiler for the RuntimeIdentifier '{0}' in the MonoAotCrossCompiler item group", RuntimeIdentifier);
-					AotCompiler = ComputeAotCompilerPath ();
+					AotCompiler = ComputeValueUsingTarget (target, targetName);
 				} else {
 					AotCompiler = aotCompilerItem.ItemSpec;
 				}
 			} else {
-				AotCompiler = ComputeAotCompilerPath ();
+				AotCompiler = ComputeValueUsingTarget (target, targetName);
 			}
 
 			if (!File.Exists (AotCompiler))
 				Log.LogError (MSBStrings.E7081 /*"The AOT compiler '{0}' does not exist." */, AotCompiler);
 
 			return !Log.HasLoggedErrors;
 		}
-
-		// If we can't find the AOT compiler path in MonoAotCrossCompiler, evaluate a project file that does know how to find it.
-		// This happens when executing remotely from Windows, because the MonoAotCrossCompiler item group will be empty in that case.
-		string ComputeAotCompilerPath ()
-		{
-			var projectPath = Path.GetTempFileName ();
-
-			File.Delete (projectPath);
-			projectPath += ".csproj";
-
-			var csproj = $@"<?xml version=""1.0"" encoding=""utf-8""?>
-<Project Sdk=""Microsoft.NET.Sdk"">
-	<PropertyGroup>
-		<TargetFramework>net{TargetFramework.Version}-{PlatformName}</TargetFramework>
-	</PropertyGroup>
-	<Target Name=""ComputeAotCompilerPath"">
-		<PropertyGroup>
-			<_XamarinAOTCompiler>@(MonoAotCrossCompiler->WithMetadataValue(""RuntimeIdentifier"", ""$(RuntimeIdentifier)""))</_XamarinAOTCompiler>
-		</PropertyGroup>
-		<WriteLinesToFile File=""$(OutputFilePath)"" Lines=""$(_XamarinAOTCompiler)"" />
-	</Target>
-</Project>
-";
-			File.WriteAllText (projectPath, csproj);
-
-			var dotnetPath = Environment.GetEnvironmentVariable ("DOTNET_HOST_PATH");
-
-			if (string.IsNullOrEmpty (dotnetPath)) {
-				dotnetPath = "dotnet";
-			}
-
-			var environment = default (Dictionary<string, string>);
-			var customHome = Environment.GetEnvironmentVariable ("DOTNET_CUSTOM_HOME");
-
-			if (!string.IsNullOrEmpty (customHome)) {
-				environment = new Dictionary<string, string> { { "HOME", customHome } };
-			}
-
-			try {
-				ExecuteRestoreAsync (dotnetPath, projectPath, environment).Wait ();
-
-				return ExecuteBuildAsync (dotnetPath, projectPath, environment).Result;
-			} finally {
-				if (KeepTemporaryOutput) {
-					Log.LogMessage (MessageImportance.Normal, $"Temporary project for the FindAotCompiler task: {projectPath}");
-				} else {
-					File.Delete (projectPath);
-				}
-			}
-		}
-
-		async Threading.Task ExecuteRestoreAsync (string dotnetPath, string projectPath, Dictionary<string, string> environment)
```

Comment:
You've removed the separate restore, and there was a reason for it to be separate: https://github.com/xamarin/xamarin-macios/commit/6bc66bf2ee5a7bd04bedde370011a7ca3eb789ee

Reply:
Done! Thanks for catching this one!